T.T type and Display Name

retothuerer opened the pull request:
Added specification for T.T type
Added

Commit:
using same example as in PAC-ID repor

Commit:
introduced Table segment

Commit:
nicer look of  railroad diagram

Commit:
restructure text
- example per segment type
- more description on table segment

retothuerer committed:
ebfn updated
- alignment of variable names
- fix bug that table header was missing ":"

retothuerer committed:
standard terms formatting per Danis recommendation
- no underlines
- in ` ` fro gray background

retothuerer committed:
corrected railroad diagram
- ":" separator in table header was missing

retothuerer committed:
background color of railroad diagram

retothuerer committed:
- Added specification for 'T.T' type
- added key for display name
- added key for production date
- specified key for TREX extension type
- cosmetic improvements

retothuerer commented on README.md:
```diff
@@ -104,20 +161,26 @@ The following section contains the grammar for the T-REX format in [EBNF](https:
 numericvalue = decimal | scientific ;
 texttype     = "T.A";
 textvalue    = { punctuation | alphanumeric };
+extendedtexttype     = "T.T";
```
@apinilabs-manuel  do you have a better suggestion to name this. I'm not convinced by this, but didn't have a better idea

retothuerer commented on README.md:
```diff
@@ -67,35 +134,25 @@ The type MUST either be a `Unit of Measure Common Code` or a hint to a data type
+| `T.A` | T-REX (this specification) | For alphanumeric strings of variable length, limited to the character set `A-Z`, `0-9`, `.` and `-`. Example: `METHOD$T.A:HELLOWORLD` |
+| `T.T` | T-REX (this specification) | For strings of variable length without limitations to the character set [^3]. <br> The string MUST first be encoded in UTF-8 and the resulting bytes be converted to [Base36](https://en.wikipedia.org/wiki/Base36) with the alphabet `01234567890ABCDEFGHIJKLMNOPQRSTUVWXYZ`. <br><br> Use this if it is desirable to use characters which are outside the character se ...
+| `T.X` | T-REX (this specification) | For arbitrary [Base36](https://en.wikipedia.org/wiki/Base36) encoded data. The alphabet is `01234567890ABCDEFGHIJKLMNOPQRSTUVWXYZ`. Use this as a last resort only. |
 | `E` | T-REX (this specification) | For error codes (alphanumeric strings of a variable length, limited to the character set `A-Z`, `0-9`, `.` and `-`). This type is meant to be used to indicate errors for expected `key`s, e.g. if a `TEMP$KEL` is not available because the corresponding sensor was unplugged, `TEMP$T.E:NC` could be used. |
```
@apinilabs-manuel is it clear enough what the difference between T.T and T.X is?

apinilabs-manuel replied:
it's mainly T.T that is misleading and should get renamed as the extremely limited characterset is not immediately clear. For the description of T.X mention early that this is base36 encoded (as people usually are lazy, so most relevant and not obvious facts first.

retothuerer committed:
Improved description of T.T type
changed naming in EBNF:
- T.A from texttype to  alphanumerictype
- T.T from extendedtexttype to texttype
